src/zarr/errors.py

```diff
@@ -34,6 +35,10 @@ class GroupNotFoundError(BaseZarrError, FileNotFoundError):
     _msg = "No group found in store {!r} at path {!r}"
 
 
+class ArrayNotFoundError(BaseZarrError):
```

Comment:
This should also sub-class FileNotFoundError to make sure we retain backwards compatibility.